sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -406,9 +406,10 @@ abstract class GpuBroadcastExchangeExecBase(
   @transient
   protected val timeout: Long = SQLConf.get.broadcastTimeout
 
-  val _runId: UUID = UUID.randomUUID()
-
-  override def runId: UUID = _runId
+  // prior to Spark 3.5.0, runId is defined as `def` rather than `val` so
+  // produces a new ID on each reference. We capture the value once and
+  // cache it here
+  val _runId: UUID = runId
```

Comment:
I don't think this does what we want prior to Spark 3.5.  This removed the `runId` override, so now when Spark calls runId on this, it will return a new ID each time (due to the default implementation of `runId` that is no longer overridden).  Scala allows `val` to override `def`, so I think we just want this, and update the calls to _runId to just use runId directly.
```suggestion
  override val runId: UUID = UUID.randomUUID
```